### ISSUE TITLE
Adding functionality to obtain inherited types

### ISSUE THREAD

**Issue by 766F6964** [Washi1337/AsmResolver#40]
Currently, there doesn't seem to be a quick way to obtain all types, that one type inherits from.
Similar libraries to AsmResolver, such as [dnlib](https://github.com/0xd4d/dnlib) provide such functionality.
An example is shown below:

```csharp
type.InheritsFrom("System.Configuration.SettingsBase"){
    Console.WriteLine("Inherits!");
}
```

Is it possible to add such functionality to AsmResolver?

**Comment by Washi1337** [comment]
There is already support for checking the super class of a type:
```csharp
if (type.BaseType != null && type.BaseType.IsTypeOf("System.Configuration", "SettingsBase"))
{
   // type is a subclass of System.Configuration.SettingsBase
}
```

Perhaps you are looking for some method that recursively goes up the type hierarchy tree, e.g. FileStream indirectly inherits from System.Object (Object -> MarshalByRefObject -> Stream -> FileStream).

How about interfaces? Should that be included as well in the search? Technically classes is not inheriting from an interface, but can be useful nonetheless. Are you looking for an AsmResolver equivalent to [Type.IsAssignableFrom](https://docs.microsoft.com/en-us/dotnet/api/system.type.isassignablefrom?view=netframework-4.8)?

**Comment by 766F6964** [comment]
Yes, I was referring to the first scenario you mentioned, indirect inheritance, by traversing the hierarchy tree. Basically, an equivalent to [this](https://github.com/yck1509/ConfuserEx/blob/3c9c29d9daf2f1259edf69054c5693d5d225a980/Confuser.Core/DnlibUtils.cs#L152-L163) function.

Interfaces should not be included in the search by default. Either provide the function with a parameter to include or exclude interfaces from the search, or make it a separate function, like [here](https://github.com/yck1509/ConfuserEx/blob/3c9c29d9daf2f1259edf69054c5693d5d225a980/Confuser.Core/DnlibUtils.cs#L171-L184).

**Comment by zsr2531** [comment]
@Washi1337 Issue can be closed, feature implemented in #81.
